### PR TITLE
feat(core): validate connector config before enabling

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Check connector config before enabling.

## Linear Issue Reference

LOG-2044

## Testing

Passed UTs.

### CODE REVIEW

File: packages/core/src/routes/connector.test.ts [hunk 1]
```diff
@@ -199,12 +199,15 @@ describe('connector route', () => {
             description: {},
             readme: 'README.md',
           },
+          // eslint-disable-next-line @typescript-eslint/no-empty-function
+          validateConfig: async (_config) => {},
```
Comment (wangsijie) [inline packages/core/src/routes/connector.test.ts:203]: jest.fn()

File: packages/core/src/routes/connector.test.ts [hunk 2]
```diff
@@ -223,6 +226,34 @@ describe('connector route', () => {
...
+          },
+          metadata: {
+            id: 'connector_0',
+            type: ConnectorType.Social,
+            name: {},
+            logo: './logo.png',
+            description: {},
+            readme: 'README.md',
+          },
+          validateConfig: async (_config) => {
```
Comment (wangsijie) [inline packages/core/src/routes/connector.test.ts:246]: `_config` can be removed

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/routes/connector.test.ts [hunk 3]
```diff
@@ -199,12 +199,15 @@ describe('connector route', () => {
...
           },
+          // eslint-disable-next-line @typescript-eslint/no-empty-function
+          validateConfig: async (_config) => {},
         };
       });
       const response = await connectorRequest
         .patch('/connectors/connector_0/enabled')
         .send({ enabled: true });
-      expect(updateConnector).toHaveBeenCalledWith(
+      expect(updateConnector).toHaveBeenNthCalledWith(
```
Comment (simeng-li) [inline packages/core/src/routes/connector.test.ts:209]: just haveBeenCalledWith is good enough?

**Review by simeng-li — APPROVED** [review]
over all LGTM

**Pull request merged**